src/sc/sidechain.cpp

```diff
@@ -503,4 +506,126 @@ bool CSidechain::GetCeasingCumTreeHash(CFieldElement& ceasedBlockCum) const
     return true;
 }
 
+#if 0
+static int getInitNumBlocksForScFeeCheck()
+#else
+int CSidechain::getNumBlocksForScFeeCheck()
+{
+    if ( (Params().NetworkIDString() == "regtest") )
+    {
+        int val = (int)(GetArg("-blocksforscfeecheck", Params().ScNumBlocksForScFeeCheck() ));
+        if (val >= 0)
+        {
+            LogPrint("sc", "%s():%d - %s: using val %d \n", __func__, __LINE__, Params().NetworkIDString(), val);
+            return val;
+        }
+        LogPrint("sc", "%s():%d - %s: val %d is negative, using default %d\n",
+            __func__, __LINE__, Params().NetworkIDString(), val, Params().ScNumBlocksForScFeeCheck());
+    }
+    return Params().ScNumBlocksForScFeeCheck();
+}
+#endif
+
+#if 0
+int CSidechain::getNumBlocksForScFeeCheck()
+{
+    // gets constructed just one time
+    static int retVal( getInitNumBlocksForScFeeCheck() );
+    return retVal;
+}
+#endif
+
+void CSidechain::InitScFees()
+{
+    // only in the very first time, calculate the size of the buffer
+    if (sizeOfScFeesContainers == -1)
+    {
+        int epochLength = fixedParams.withdrawalEpochLength;
+
+        assert(epochLength > 0);
+       
+        int numBlocks = getNumBlocksForScFeeCheck();
+        sizeOfScFeesContainers = numBlocks / epochLength;
+        if (sizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
+        {
+            sizeOfScFeesContainers++;
+        }
+        LogPrint("sc", "%s():%d - sizeOfScFeesContainers set to %d\n", __func__, __LINE__, sizeOfScFeesContainers);
+
+        assert(sizeOfScFeesContainers > 0);
+        assert(scFees.empty());
+        scFees.resize(sizeOfScFeesContainers);
```

Comment:
Should we use the FT fee and MBTR fee values to initialize the `scFees` list instead of 0? I guess that by using 0 it could be possible to include FT or MBTR with 0 fees in a block during the first N epochs, since we have a different check for transactions in `AcceptTxToMemoryPool()` and in `ConnectBlock()`.

I see two possible solutions:

- initializing the scFees list with the N values equal to the fees declared during sidechain creation

- initializing the scFees list with one single value (equal to the fees declared during sidechain creation) and gradually increasing the size of the list until we reach the epoch number N (and the list is full)

Reply:
Agree; I chose the first option. I also moved the InitScFees() method call into sc creation handling

Reply:
After further discussion, the second approach has been choosen.